AUTH: add password strength bar

cviveiro opened the pull request:
Works on some of the frontend parts of https://github.com/MetLifeLegalPlans/portunus/issues/169

Many questions/comments:
1) I took out the password help text because it looks pretty horrible with the bar, figured that the error message will be enough (and the fact that numbers and letters in combination will also make the meter move towards stronger). If we want them to work together, or indeed to make this look better, it doesn't look that hard to roll out our own component, which could do some more specific checks, put in helpful hints with the bar (I didn't find any components out of the box that do this but the underlying library provides this in it's result value, so... maybe I'm missing a possible usable open-source component?), match what we want it to look like, etc.
![screencapture-localhost-3001-register-2020-08-21-09_38_25](https://user-images.githubusercontent.com/5933912/90921095-6231f300-e39e-11ea-9223-18eb2cfeb1d4.png)
2) We're not doing anything based on the level of weakness. I was able to register with a weak password. Do we want to use this to make sure the password is strong based on this? If we did so, we'd likely want to implement something on the backend too but that shouldn't be bad. The method this uses via [zxcvbn](https://github.com/dropbox/zxcvbn) does look like a legit way of measuring whether a password is strong or not, unlike the whole, must-contain-these-specific-classes-of-character-but-we're-not-looking-at-much-else philosophy.
3) Do we even want this? Or should we push back? I did this because it looked lightly interesting and easy if we pick one out of the box, but this looked like a (strong) recommendation on BofA's part.

cviveiro commented on frontend/constants.js:
```diff
@@ -11,3 +11,5 @@ export const PRIVACY_REQUEST_URL = 'https://www.metlife.com/privacy-request';
 
 export const SUPPORT_PHONE_NUMBER = '800-821-6400';
+
+export const MIN_PASSWORD_LENGTH = 8;
```
Agreed, I'm saving this for a beginner django PR for Pooja

markrawls commented:
> The method this uses via zxcvbn does look like a legit way of measuring whether a password is strong or not, unlike the whole, must-contain-these-specific-classes-of-character-but-we're-not-looking-at-much-else philosophy.

That's an awesome library with a really bad name. I love it! That'd make it easy to enforce these same set of standards on the backend as well, which I think we should do.

> I didn't find any components out of the box that do this but the underlying library provides this in it's result value, so... maybe I'm missing a possible usable open-source component

Most forms I've seen put a little question mark icon with a tooltip that pops up with what you need to know. There's a `Tooltip` component in `wui` that just wraps the base [material-ui toolkit](https://material-ui.com/components/tooltips/#simple-tooltips) and gives a default icon.

cviveiro commented:
Whoops, forgot to comment. This is updated and ready for review. I've added in a tooltip to show hints as to what will help (if anyone has improvements on what to say they're greatly appreciated) and I've added in a restriction on the score allowed.

markrawls reviewed (approved):
Looks fine to me! I still think this should be enforced on the backend as well though.

cviveiro commented:
Definitely agree about the backend verification, I'm saving it for a beginner independent project
